### PR TITLE
sipify gui part 5

### PR DESCRIPTION
**Pull request by 3nids**
## Description
Include a few sentences describing the overall goals for this PR (pull request). If applicable also add screenshots.

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [ ] Commit messages are descriptive and explain the rationale for changes
- [ ] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [ ] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [ ] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and containt sufficient information in the commit message to be documented
- [ ] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] New unit tests have been added for core changes
- [ ] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTE.md#contributing-to-qgis) before each commit

### CODE REVIEW

File: python/gui/qgsmaplayerconfigwidgetfactory.sip [hunk 1]
```diff
@@ -1,89 +1,124 @@
...
-     * @return True if this layer is supported for this widget
-     */
     virtual bool supportsLayer( QgsMapLayer *layer ) const;
+%Docstring
+ Check if the layer is supported for this widget.
+ :return: True if this layer is supported for this widget
+ :rtype: bool
+%End
+
+    virtual QgsMapLayerConfigWidget *createWidget( QgsMapLayer *layer, QgsMapCanvas *canvas, bool dockWidget = true, QWidget *parent = 0 ) const = 0;
```
Comment (nyalldawson) [inline python/gui/qgsmaplayerconfigwidgetfactory.sip]: Missing Factory

File: python/gui/qgsmapmouseevent.sip [hunk 2]
```diff
@@ -1,118 +1,140 @@
...
-%TypeHeaderCode
-#include "qgsmapmouseevent.h"
+%Docstring
+ A QgsMapMouseEvent is the result of a user interaction with the mouse on a QgsMapCanvas.
+ It is sent whenever the user moves, clicks, releases or double clicks the mouse.
+ In addition to the coordiantes in pixel space it also knows the coordinates in the mapcanvas' CRS
+ as well as it knows the concept of snapping.
 %End
 
-%ConvertToSubClassCode
```
Comment (nyalldawson) [inline python/gui/qgsmapmouseevent.sip]: Missing subclassing

File: python/gui/qgsmapoverviewcanvas.sip [hunk 3]
```diff
@@ -1,63 +1,130 @@
...
 %End
 
+%TypeHeaderCode
+#include "qgsmapoverviewcanvas.h"
+%End
   public:
-    QgsMapOverviewCanvas( QWidget *parent /TransferThis/ = 0, QgsMapCanvas *mapCanvas = NULL );
-
-    ~QgsMapOverviewCanvas();
+    QgsMapOverviewCanvas( QWidget *parent = 0, QgsMapCanvas *mapCanvas = 0 );
```
Comment (nyalldawson) [inline python/gui/qgsmapoverviewcanvas.sip]: TransferThis

**Comment by 3nids** [comment]
@nyalldawson thanks a lot for the reviews,  but these were quick tries, I just removed them from black list to see what was missing. 
Thanks for doing it.

**Pull request merged**